pkg/node/node.go

```diff
@@ -1076,25 +1076,29 @@ func (s *Service) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest)
 						log.Errorf("couldn't get targets from array: %s", err.Error())
 						continue
 					}
-
-					log.Infof("Discovering NVMeTCP targets")
-					nvmetcpConnectCount := 0
-					nvmeIP := strings.Split(infoList[0].Portal, ":")
-					nvmeTargets, err := s.nvmeLib.DiscoverNVMeTCPTargets(nvmeIP[0], false)
-					if err != nil {
-						log.Error("couldn't discover NVMe targets")
-						continue
-					} else {
-						for _, target := range nvmeTargets {
-							err = s.nvmeLib.NVMeTCPConnect(target, false)
-							if err != nil {
-								log.Infof("couldn't connect to NVMeTCP targets")
-							} else {
-								nvmetcpConnectCount = nvmetcpConnectCount + 1
-							}
+					var nvmeTargets []gonvme.NVMeTarget
+					for _, address := range infoList {
+						// doesn't matter how many portals are present, discovering from any one will list out all targets
+						nvmeIP := strings.Split(address.Portal, ":")
+						log.Info("Trying to discover NVMe target from portal ", nvmeIP[0])
+						nvmeTargets, err = s.nvmeLib.DiscoverNVMeTCPTargets(nvmeIP[0], false)
+						if err != nil {
+							log.Error("couldn't discover targets")
+							continue
 						}
+						break
```

Comment:
In the `infoList` if you are able to discover one item you are exiting the for loop continues with the discovered nvme targets  and in the line, no's: [1092-1100](https://github.com/dell/csi-powerstore/pull/196/files#diff-515b2d0691202c4ea5e578ec896258ae9c440c381b608ed1eb7116a27847057dR1092-R1100)
 you are trying to connect to the discovered targets, what if we are not able to connect that targets as we have not iterated to the full `infoList` we will miss some of the targets to which we are able to connect.

Thank You

Reply:
@satyakonduri There is no need to discover targets from all addresses here. Discovering from one address is enough since one success discovery response will contain all target points from all networks.
Just to make our code optimize and error free we will try to iterate again only if we have not discovered the target from one address else break and no need to discover the same targets.

Reply:
Thank You